scheduler.py

```diff
@@ -71,3 +76,27 @@ def get_task_completion(self, task):
     def sigterm_handler(self, signum, frame):
         logging.info("Exiting gracefully.")
         sys.exit(0)
+
+    def get_worker(self):
+        if self.schedulerMode == constants.SCHEDULINGMODE_RANDOM:
+            logging.info(f"Getting worker for {self.schedulerMode}")
+            if self.globalIncrementalWorkerId > 1:
+                random_worker_id = random.randint(0, self.globalIncrementalWorkerId - 1)
+                with self.workerMapLock:
+                    return self.workers[random_worker_id]
+            else:
+                logging.info(f"No worker found.")
+                raise NoWorkerAvailableError(self.schedulerMode)
+        elif self.schedulerMode == constants.SCHEDULINGMODE_ROUNDROBIN:
+            logging.info(f"Getting worker for {self.schedulerMode}")
+            if self.workerQueue.qsize() > 0:
+                # dequeue and enqueue again and return the wrapped worker info object
+                wrappedWorkerInfo = self.workerQueue.get()
```

Comment:
why aren't we using lock here? Lines 92-95 will be shared among multiple threads